### ISSUE TITLE
latest plugin update v3.1.10 breaks site, missing dependancy

### ISSUE THREAD

**Issue by eballeste** [joppuyo/acf-image-aspect-ratio-crop#19]
PHP Fatal error: Uncaught Error: Class 'Ramsey\\Uuid\\Uuid' not found in [redacted]/wp-content/plugins/acf-image-aspect-ratio-crop/fields/class-npx-acf-field-image-aspect-ratio-crop-v5.php:82\nStack trace:\n#0

**Comment by eballeste** [comment]
I updated from v3.0.1 to v3.1.10

**Comment by joppuyo** [comment]
Hey, could you tell me the version of the plugin and how you installed it (wordpress.org plugin repository, Composer, zip from GitHub releases)

There were a few problems with a recent version since I changed to the deployment process from Travis to GitHub actions but it should be fixed in the latest version

**Comment by eballeste** [comment]
I had 3.0.1 and updated via the plugins panel directly to 3.1.10

**Comment by eballeste** [comment]
not sure if this helps but I had as a setting for all of these fields to return just the image ID

**Comment by eballeste** [comment]
I had to revert to the old version because my deadline is today so I won't be able to do further tests :(

**Comment by joppuyo** [comment]
Hey, apologies this is causing you issues. I'd still like to figure out what is causing this issue in case other users may run into it.

Could you tell me what PHP version you are running?

**Comment by eballeste** [comment]
php 7.3.8

**Comment by eballeste** [comment]
all of these fields are setup to return only the ID, as soon as I installed the plugin, the front-end crashed with the error shown above, `Ramsey\Uuid\Uuid` is missing

**Comment by eballeste** [comment]
did you maybe change something with how your composer dependancies are loaded? could it be that it's somehow now conflicting with my composer settings and it can't find your vendor files?? just throwing wild guesses out there, I noticed that the library is present in both versions of the plugin

**Comment by eballeste** [comment]
- Uncaught Error: Class 'Ramsey\Uuid\Uuid' not found in /[redacted]/wp-content/plugins/acf-image-aspect-ratio-crop/fields/class-npx-acf-field-image-aspect-ratio-crop-v5.php:82
Stack trace:
- #0 /[redacted]/wp-content/plugins/acf-image-aspect-ratio-crop/fields/class-npx-acf-field-image-aspect-ratio-crop-v5.php(1052): npx_acf_field_image_aspect_ratio_crop-&gt;__construct(Array)
- #1 /[redacted]/wp-content/plugins/acf-image-aspect-ratio-crop/acf-image-aspect-ratio-crop.php(464): include_once('/Users/division...')
- #2 /[redacted]/wp-includes/class-wp-hook.php(288): npx_acf_plugin_image_aspect_ratio_crop-&gt;include_field_types(5)
- #3 /[redacted]/wp-includes/class-wp-hook.php(312): WP_Hook-&gt;apply_filters(NULL, Array)
- #4 /[redacted]/wp-includes/plugin.php(478): WP_Hook-&gt;do_action(Array)
- #5 /[redacted]/wp-content/plugins/acf-image-aspect-ratio-crop/fields/class-npx-acf-field-image-aspect-ratio-crop-v5.php on line 82

**Comment by eballeste** [comment]
this is the new line that's causing the break:
`$this->temp_post_id = Ramsey\Uuid\Uuid::uuid4()->toString();`

for some reason the composer auto-loader isn't kicking in

**Comment by joppuyo** [comment]
Yeah, I think there is some kind of conflict with Composer. Could you tell me how are you using Composer in your theme? Do you use Composer to install WordPress (like [Bedrock](https://roots.io/bedrock/)) or are you using Composer in your theme?

**Comment by eballeste** [comment]
i have composer and the vendor folder in the root of the project, this installs plugins (yours was not installed this way and is working fine in v.3.0.1) and also library packages used in the theme

i noticed that in your codebase, previous code that works that uses the same Ramsey library works fine:

```
use Ramsey\Uuid\Uuid;

// exit if accessed directly
if (!defined('ABSPATH')) {
    exit();
}

// Check if we are using local Composer
if (file_exists(__DIR__ . '/vendor')) {
    require 'vendor/autoload.php';
}
```

you are not doing the same in `class-npx-acf-field-image-aspect-ratio-crop-v5.php`

**Comment by joppuyo** [comment]
This shouldn't be a problem since the plugin file loaded first and it includes the composer autoloader. The class for the field is loaded after this inside [a hook](https://github.com/joppuyo/acf-image-aspect-ratio-crop/blob/9f269fd505705a7b5acc32edf1a1b9716800c4cf/acf-image-aspect-ratio-crop.php#L72)

Could you check that the `acf-image-aspect-ratio-crop/vendor/autoload.php` is physically present on the server? I have personally had some problems with deploying wp sites that have `vendor` listed in .gitignore file instead of `/vendor` since not having the first slash will cause **all plugins** with `vendor` folder in the project break.

I'm thinking that this issue may have always existed on your site since previously the Composer uuid and guzzle libraries were called conditionally [here](https://github.com/joppuyo/acf-image-aspect-ratio-crop/blob/9f269fd505705a7b5acc32edf1a1b9716800c4cf/acf-image-aspect-ratio-crop.php#L209) and [here](https://github.com/joppuyo/acf-image-aspect-ratio-crop/blob/9f269fd505705a7b5acc32edf1a1b9716800c4cf/acf-image-aspect-ratio-crop.php#L213) so if you were not cropping remote images you might not have ran into a code path that triggers this issue at all.

**Comment by joppuyo** [comment]
Hey, I have released version 3.1.11 of the plugin which uses the native wp_generate_uuid4() function instead of the Composer dependency.

Please test it out and tell me if it fixes the issue.

I will have to try to further debug the cause of this issue but I think now the most important thing in to release a version of this plugin that works and does not break people's sites.

Thanks!
